### DISCUSSION TITLE
Git tag regex for terraform manager -> module bug

### DISCUSSION THREAD

**Discussion by Alphaegen** [renovatebot/renovate#32242] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab

### Please tell us more about your question or problem

I've got the following two private terraform modules:
- git@gitlab.x.nl:terraform-modules-x/monitoring/azure-zabbix.git?ref=1.2.10
- git@gitlab.x.nl:terraform-modules-x/azure-grafana.git?ref=1.23.10

The problem is that the following code ([location](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/terraform/extractors/others/modules.ts#L18C1-L20C3)) doesn't support the second link as it's missing a second `/`:
```
export const gitTagsRefMatchRegex = regEx(
  /(?:git::)?(?<url>(?:(?:http|https|ssh):\/\/)?(?:.*@)?(?<path>.*\/(?<project>.*\/.*)))\?(depth=\d+&)?ref=(?<tag>.*?)(&depth=\d+)?$/,
);
```

Could this maybe be changed to allow one less `/`, maybe by doing this? I added a `*` to the second `/` to make it optional.
```
export const gitTagsRefMatchRegex = regEx(
  /(?:git::)?(?<url>(?:(?:http|https|ssh):\/\/)?(?:.*@)?(?<path>.*\/(?<project>.*\/*.*)))\?(depth=\d+&)?ref=(?<tag>.*?)(&depth=\d+)?$/,
);
```

This would mean the package name will change, but I'm just suggesting a possible solution.

### Logs (if relevant)

I don't think logs are relevant, but I can add the following relevant regex101 links:

Currently: [solution](https://regex101.com/r/OSvr9V/1)
My proposed solution: [solution](https://regex101.com/r/ApWEcX/1)

**Comment by rarkins** [comment]
What would be some example package name changes? packageName or depName?

Cc @secustor

**Reply by Alphaegen** [reply]
As you can see in the solution links, the `project` group becomes: `azure-zabbix` instead of `monitoring/azure-zabbix`. It looks like the `monitoring` is being filtered out anyway tho at [code](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/terraform/extractors/others/modules.ts#L91)

**Reply by rarkins** [reply]
Can you give a before/after of how that would look for the `packageFiles with updates` log message?

**Comment by Alphaegen** [comment]
I'm afraid that'll be difficult with our setup, as we're using the RenovateBot docker image within a private cloud to which I personally don't have access to. Only the RenovateBot does.

**Reply by Alphaegen** [reply]
As in it'll be difficult to make a local change to the code of RenovateBot with our use-case.
